### PR TITLE
Change "classification_model" to "model"

### PR DESCRIPTION
**Pull request by nilsleh**
Change the argument name for the classifcation task according to comment in #854.

I put this in a separate PR because with classification there are quiet a lot of file changes.

### CODE REVIEW

File: evaluate.py [hunk 1]
```diff
@@ -163,18 +163,18 @@ def main(args: argparse.Namespace) -> None:
     if issubclass(TASK, ClassificationTask):
```
Comment (adamjstewart) [inline evaluate.py:163]: Need to undo chmod changes

File: evaluate.py [hunk 2]
```diff
@@ -163,18 +163,18 @@ def main(args: argparse.Namespace) -> None:
...
         val_row = {
             "split": "val",
-            "classification_model": hparams["classification_model"],
-            "learning_rate": hparams["learning_rate"],
-            "weights": hparams["weights"],
-            "loss": hparams["loss"],
+            "model": model.hparams["model"],
```
Comment (adamjstewart) [inline evaluate.py:166]: Bad rebase, these should all be `hparams`, not `model.hparams` after my Lightning PR.

File: torchgeo/trainers/classification.py [hunk 3]
```diff
@@ -108,13 +106,13 @@ def __init__(self, **kwargs: Any) -> None:
...
 
         Keyword Args:
-            classification_model: Name of the classification model use
+            model: Name of the classification model use
             loss: Name of the loss function
-            weights: Either "random" or "imagenet"
-            num_classes: Number of prediction classes
-            in_channels: Number of input channels to model
-            learning_rate: Learning rate for optimizer
-            learning_rate_schedule_patience: Patience for learning rate scheduler
```
Comment (adamjstewart) [inline torchgeo/trainers/classification.py:117]: Why remove these kwarg docs?

File: torchgeo/trainers/utils.py [hunk 4]
```diff
@@ -51,8 +50,7 @@ def extract_encoder(path: str) -> Tuple[str, "OrderedDict[str, Tensor]"]:
         )
     else:
         raise ValueError(
-            "Unknown checkpoint task. Only encoder or classification_model"
-            " extraction is supported"
+            "Unknown checkpoint task. Only encoder or model" " extraction is supported"
```
Comment (adamjstewart) [inline torchgeo/trainers/utils.py:53]:
```suggestion
            "Unknown checkpoint task. Only encoder or model extraction is supported"
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: evaluate.py (hunk 1 above)
Comment (nilsleh) [reply evaluate.py:163]: sorry, I don't understand what you mean here.

Comment (adamjstewart) [reply evaluate.py:163]: This file was previously executable (you could run `./evaluate.py` on Unix) with mode 100755, but is no longer executable with mode 100644. `chmod +x evaluate.py` should do the trick. If you're on Windows and don't know what any of this means, let me know and I can do it.

File: torchgeo/trainers/classification.py [hunk 5]
```diff
@@ -68,26 +68,24 @@ def config_model(self) -> None:
             custom_pretrained = True
 
         # Create the model
-        valid_models = timm.list_models(pretrained=True)
-        if classification_model in valid_models:
+        valid_models = timm.list_models(pretrained=False)
```
Comment (adamjstewart) [inline torchgeo/trainers/classification.py:71]:
```suggestion
        valid_models = timm.list_models(pretrained=imagenet_pretrained)
```

File: tests/trainers/test_utils.py [hunk 6]
```diff
@@ -56,10 +56,7 @@ def test_extract_encoder_unsupported_model(tmp_path: Path) -> None:
     checkpoint = {"hyper_parameters": {"some_unsupported_model": "resnet18"}}
     path = os.path.join(str(tmp_path), "dummy.ckpt")
     torch.save(checkpoint, path)
-    err = (
-        "Unknown checkpoint task. Only encoder or classification_model"
-        " extraction is supported"
-    )
+    err = "Unknown checkpoint task. Only encoder or model" " extraction is supported"
```
Comment (adamjstewart) [inline tests/trainers/test_utils.py:59]:
```suggestion
    err = "Unknown checkpoint task. Only encoder or model extraction is supported"
```

**Review by adamjstewart — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]

**Review by ashnair1 — APPROVED** [review]

**Pull request merged**